Sources/SwiftDocCTestUtilities/SymbolGraphCreation.swift

```diff
@@ -12,28 +12,167 @@
     ) -> SymbolGraph {
         return SymbolGraph(
-            metadata: SymbolGraph.Metadata(
-                formatVersion: SymbolGraph.SemanticVersion(major: 0, minor: 6, patch: 0),
-                generator: "unit-test"
-            ),
-            module: SymbolGraph.Module(
-                name: moduleName,
-                platform: platform
-            ),
+            metadata: makeMetadata(),
+            module: makeModule(moduleName: moduleName, platform: platform),
             symbols: symbols,
             relationships: relationships
         )
     }
     
+    package func makeMetadata(major: Int = 0, minor: Int = 6, patch: Int = 0) -> SymbolGraph.Metadata {
+        SymbolGraph.Metadata(
+            formatVersion: SymbolGraph.SemanticVersion(major: major, minor: minor, patch: patch),
+            generator: "unit-test"
+        )
+    }
+    
+    package func makeModule(moduleName: String, platform: SymbolGraph.Platform = .init()) -> SymbolGraph.Module {
+        SymbolGraph.Module(name: moduleName, platform: platform)
+    }
+    
+    // MARK: Line List
+    
+    package func makeLineList(
+        docComment: String,
+        startOffset: SymbolGraph.LineList.SourceRange.Position = defaultSymbolPosition,
+        url: URL = defaultSymbolURL
+    ) -> SymbolGraph.LineList {
+        SymbolGraph.LineList(
+            // Create a `LineList/Line` for each line of the doc comment and calculate a realistic range for each line.
+            docComment.components(separatedBy: .newlines)
+                .enumerated()
+                .map { lineOffset, line in
+                    SymbolGraph.LineList.Line(
+                        text: line,
+                        range: SymbolGraph.LineList.SourceRange(
+                            start: .init(line: startOffset.line + lineOffset, character: startOffset.character),
+                            end:   .init(line: startOffset.line + lineOffset, character: startOffset.character + line.count)
+                        )
+                    )
+                },
+            // We want to include the file:// scheme here
+            uri: url.absoluteString
+        )
+    }
+    
+    package func makeMixins(_ mixins: [any Mixin]) -> [String: any Mixin] {
+        [String: any Mixin](
+            mixins.map { (type(of: $0).mixinKey, $0) },
+            uniquingKeysWith: { old, _ in old /* Keep the first encountered value */ }
+        )
+    }
+    
+    // MARK: Symbol
+    
+    package func makeSymbol(
+        id: String,
+        language: SourceLanguage = .swift,
+        kind kindID: SymbolGraph.Symbol.KindIdentifier,
+        pathComponents: [String],
+        docComment: String? = nil,
+        accessLevel: SymbolGraph.Symbol.AccessControl = .init(rawValue: "public"), // Defined internally in SwiftDocC
+        location: (position: SymbolGraph.LineList.SourceRange.Position, url: URL)? = (defaultSymbolPosition, defaultSymbolURL),
```

Comment:
We're duplicating these defaults 3 times in this file, it would be nice if we only used them once. I think here we could make it a non-optional value and save us needing to have defaults later on:
https://github.com/apple/swift-docc/blob/64ec387cab27ce898f74df36eff2d0d548b77658/Sources/SwiftDocCTestUtilities/SymbolGraphCreation.swift#L108-L109

```suggestion
        location: (position: SymbolGraph.LineList.SourceRange.Position, url: URL) = (defaultSymbolPosition, defaultSymbolURL),
```

and then we won't need the defaults:
```swift
makeLineList(
    docComment: $0,
    startOffset: location.position,
    url: location.url
)
```

Reply:
There are some cases where real world data doesn't have a location, for example if the symbol graph was extracted from a binary instead of from source code. 

I prefer to keep the ability to create such a value.